### PR TITLE
[IOTDB-325] Refactor Statistics

### PR DESCRIPTION
**Pull request by qiaojialin**
Move start time, end time, count in PageHeader and ChunkMetadata into Statistics inside the two
Remove maxTombstoneTime in ChunkHeader
Remove StatisticsForFilter
simplify the serialize and deserialize of Statistics

### CODE REVIEW

**Comment by jixuan1989** [comment]
can you point out your detailed changes?

**Comment by qiaojialin** [comment]
> can you point out your detailed changes?

in the pr description

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/ChunkMetaData.java [hunk 1]
```diff
@@ -78,90 +68,23 @@ private ChunkMetaData() {
...
-                    : valuesStatistics.getDigestSerializedSize()));
-    serializedSize += measurementUid.getBytes(TSFileConfig.STRING_CHARSET).length;  // measurementUid
-    return serializedSize;
+    this.statistics = statistics;
   }
 
   @Override
   public String toString() {
-    return String.format("numPoints %d", numOfPoints);
+    return String.format("numPoints %d", statistics.getCount());
```
[the remark this answers is not in the document]
Comment (qiaojialin) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/ChunkMetaData.java]: printed

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/statistics/Statistics.java [hunk 2]
```diff
@@ -152,433 +149,277 @@ public void mergeStatistics(Statistics<?> stats) {
...
+    if (time[batchSize-1] > this.endTime) {
+      endTime = time[batchSize-1];
+    }
+    count += batchSize;
+    updateStats(values, batchSize);
   }
 
-  public void updateStats(float value) {
-    throw new UnsupportedOperationException();
+  public void update(long[] time, int[] values, int batchSize) {
```
[the remark this answers is not in the document]
Comment (qiaojialin) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/statistics/Statistics.java]: added

File: tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java [hunk 3]
```diff
@@ -192,6 +192,18 @@ public static int write(int n, OutputStream outputStream) throws IOException {
     return bytes.length;
   }
 
+  public static int write(byte[] bytes, OutputStream outputStream) throws IOException {
+    outputStream.write(bytes);
+    return bytes.length;
+  }
+
+  public static int write(Binary binary, OutputStream outputStream) throws IOException {
```
[the remark this answers is not in the document]
Comment (qiaojialin) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java]: added

File: tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java [hunk 4]
```diff
@@ -192,6 +192,18 @@ public static int write(int n, OutputStream outputStream) throws IOException {
     return bytes.length;
   }
 
+  public static int write(byte[] bytes, OutputStream outputStream) throws IOException {
```
[the remark this answers is not in the document]
Comment (qiaojialin) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java]: removed

File: tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java [hunk 5]
```diff
@@ -571,6 +583,8 @@ public static ByteBuffer readByteBufferWithSelfDescriptionLength(InputStream inp
   /**
    * read bytes from byteBuffer, this method makes sure that you can read length bytes or reach to
    * the end of the buffer.
+   *
+   * read a int + buffer
    */
   public static ByteBuffer readByteBufferWithSelfDescriptionLength(ByteBuffer buffer) {
```
[the remark this answers is not in the document]
Comment (qiaojialin) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/utils/ReadWriteIOUtils.java]: added

**Pull request merged**